packages/browser-wallet/src/popup/pages/Setup/PerformRecovery.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from 'react';
+import PageHeader from '@popup/shared/PageHeader';
+import { useNavigate } from 'react-router-dom';
+import { useAtom, useAtomValue } from 'jotai';
+import { useTranslation } from 'react-i18next';
+import { networkConfigurationAtom, seedPhraseAtom } from '@popup/store/settings';
+import { popupMessageHandler } from '@popup/shared/message-handler';
+import { InternalMessageType } from '@concordium/browser-wallet-message-hub';
+import { JsonRpcClient, HttpProvider } from '@concordium/web-sdk';
+import { identitiesAtom, identityProvidersAtom } from '@popup/store/identity';
+import PendingArrows from '@assets/svg/pending-arrows.svg';
+import Button from '@popup/shared/Button';
+import { absoluteRoutes } from '@popup/constants/routes';
+import { BackgroundResponseStatus } from '@shared/utils/types';
+import { getIdentityProviders } from '@popup/shared/utils/wallet-proxy';
+import { getNet } from '@shared/utils/network-helpers';
+
+// TODO: improve "error state"
+
+function DisplayRecoveryResult() {
+    const { t } = useTranslation('setup');
+    const navigate = useNavigate();
+    const identities = useAtomValue(identitiesAtom);
+
+    return (
+        <>
+            <div className="onboarding-setup__page-with-header__description">
+                {t(identities.length ? 'performRecovery.description.after' : 'performRecovery.description.noneFound')}
+            </div>
+            <div className="onboarding-setup__recovery__results">
+                {identities.map((identity) => (
+                    <div className="onboarding-setup__recovery__identity">
+                        <p>{identity.name}</p>
+                    </div>
+                ))}
+            </div>
+            <Button
+                width="narrow"
+                className="onboarding-setup__recovery__button"
+                onClick={() => navigate(absoluteRoutes.home.account.path)}
+            >
+                {t('continue')}
+            </Button>
+        </>
+    );
+}
+
+export default function PerformRecovery() {
+    const { t } = useTranslation('setup');
+    const network = useAtomValue(networkConfigurationAtom);
+    const seedPhrase = useAtomValue(seedPhraseAtom);
+    const [providers, setProviders] = useAtom(identityProvidersAtom);
+    const [result, setResult] = useState<BackgroundResponseStatus>();
+
+    useEffect(() => {
+        getIdentityProviders()
+            .then(setProviders)
+            .catch(() => setResult(BackgroundResponseStatus.Error));
```

Comment:
If I get an error, like here, then I am shown "Error" but then I can continue into the wallet without anything recovered. I am not sure if that is how we would want to do that, in a total failure case I guess the user should retry the whole thing?

Reply:
I made it so you can retry. (But didn't stop it from closing and going into the account view)